Realm.Shared/Dynamic/DynamicRealmObjectHelper.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Realms.Weaving;
+
+namespace Realms.Dynamic
+{
+    internal class DynamicRealmObjectHelper : IRealmObjectHelper
```

Comment:
Also, I'm not what's the convention here, but I'd expose the `Instance` as a property backed by a `Lazy` field.